web-app/js/portal/data/DataCollection.js

```diff
@@ -67,6 +67,16 @@ Portal.data.DataCollection = function() {
         return applicableDownloadOptions;
     };
 
+    constructor.prototype.getFilterParams = function() {
```

Comment:
What's the difference between `getFilter` and `getFilterParams`?  

Secondly, let's sit down together and review this class together because I think we should refactor it ASAP to be more like what we'd originally envisaged.